tests/test_class_schema.py

```diff
@@ -132,6 +137,33 @@ class A:
         schema = class_schema(A)()
         self.assertRaises(ValidationError, lambda: schema.load({"data": None}))
 
+    def test_literal(self):
+        @dataclasses.dataclass
+        class A:
+            data: Literal["a"]
+
+        schema = class_schema(A)()
+        self.assertEqual(A(data="a"), schema.load({"data": "a"}))
+        self.assertEqual(schema.dump(A(data="a")), {"data": "a"})
+        for data in ["b", 2, 2.34, False]:
+            self.assertRaises(
+                ValidationError, lambda data=data: schema.load({"data": data})
+            )
```

Comment:
This is not very readable. Maybe use assertRaises as a context manager ?

```python
with self.assertRaises(ValidationError):
    schema.load({"data": data})
```

?

Reply:
Good idea, I've made the change as you suggested.